tsnet/tsnet.go

```diff
@@ -528,6 +529,9 @@ func (s *Server) Listen(network, addr string) (net.Listener, error) {
 	if err != nil {
 		return nil, fmt.Errorf("tsnet: %w", err)
 	}
+	if host != "" && host != s.Hostname {
+		return nil, fmt.Errorf("tsnet: hostname %q is not supported", host)
+	}
```

Comment:
this kinda implies a hostname means something, though?

It doesn't yet, because we don't know the hostname-to-IP(s) mapping at the time of Listen... we might not yet be in control.

And if addr is an IP address, it should be possible to listen on both the Tailscale IP and IPv6 separately too.

Also: if addr is an IP, its adddress family should be checked against tcp4 or tcp6 if those are specified.

Reply:
> this kinda implies a hostname means something, though?

Yes, and I'm not sure whether it does. In `start()` the `s.Hostname` gets plumbed into the preferences, so I guess it does affect what we ask the coordination server to authenticate for? But the "host" part of the address doesn't appear to be used again until we're routing TCP sessions later on.

So maybe a better solution would be to reject _any_ non-empty "host" field here? We don't seem to make any other use of it.